Packages/Settings/Sources/SettingsUI/Internal/GitHubSettings/GitHubSettingsViewModel.swift

```diff
@@ -11,58 +11,101 @@ final class GitHubSettingsViewModel: ObservableObject {
+    @Published var selfHostedRaw: String = ""
     @Published var runnerScope: GitHubRunnerScope
     @Published private(set) var isSettingsEnabled = true
 
     private let credentialsStore: GitHubCredentialsStore
     private var cancellables: Set<AnyCancellable> = []
     private var createAppURL: URL {
-        var url = URL(string: "https://github.com")!
-        if !organizationName.isEmpty, case .organization = runnerScope {
-            url = url.appending(path: "/organizations/\(organizationName)")
+        get async {
+            var url = await credentialsStore.selfHostedURL ?? .gitHub
+            if !organizationName.isEmpty, case .organization = runnerScope {
+                url = url.appending(path: "/organizations/\(organizationName)")
+            }
+            return url.appending(path: "/settings/apps")
         }
-        return url.appending(path: "/settings/apps")
     }
 
     init(settingsStore: SettingsStore, credentialsStore: GitHubCredentialsStore, isSettingsEnabled: AnyPublisher<Bool, Never>) {
         self.settingsStore = settingsStore
         self.credentialsStore = credentialsStore
+        self.version = settingsStore.githubServiceVersion
         self.runnerScope = settingsStore.githubRunnerScope
         isSettingsEnabled.assign(to: \.isSettingsEnabled, on: self).store(in: &cancellables)
         $appId.debounce(for: 0.5, scheduler: DispatchQueue.main).nilIfEmpty().dropFirst().sink { [weak self] appId in
             Task {
                 await self?.credentialsStore.setAppID(appId)
             }
         }.store(in: &cancellables)
+
+        $version
+            .combineLatest($selfHostedRaw)
+            .debounce(for: 0.5, scheduler: DispatchQueue.main)
+            .dropFirst()
+            .sink { [weak self] version, value in
+                self?.settingsStore.githubServiceVersion = version
+                switch version {
+                case .dotCom:
+                    Task {
+                        await self?.credentialsStore.setSelfHostedURL(nil)
+                    }
+                case .enterprise:
+                    Task {
+                        await self?.credentialsStore.setSelfHostedURL(URL(string: value))
+                    }
+                }
+            }
+            .store(in: &cancellables)
+
         $runnerScope
             .combineLatest(
                 $organizationName.nilIfEmpty(),
                 $ownerName.nilIfEmpty(),
                 $repositoryName.nilIfEmpty()
             )
+            .combineLatest($enterpriseName.nilIfEmpty())
             .debounce(for: 0.5, scheduler: DispatchQueue.main)
             .dropFirst()
-            .sink { [weak self] runnerScope, organizationName, ownerName, repositoryName in
+            .map { tuple, enterpriseName in
+                let runnerScope = tuple.0
+                let organizationName = tuple.1
+                let ownerName = tuple.2
+                let repositoryName = tuple.3
```

Comment:
This seems like a good opportunity to use tuple destructuring.

```suggestion
                let (runnerScope, organizationName, ownerName, repositoryName) = tuple
```